giotto/mapper/utils/visualization.py

```diff
@@ -27,50 +28,111 @@ def set_node_sizeref(node_elements, node_scale=12):
-        return None
+def _get_column_color_buttons(data, is_data_dataframe, node_elements,
+                              node_colors_color_variable, colorscale):
+    # TODO: Consider opting for just-in-time computation instead of computing
+    #  all node summary values ahead-of-time. Solution should preserve scroll
+    #  zoom functionality of 2D static visualisation.
+    if is_data_dataframe:
+        columns_to_color = data.columns
     else:
-        column_color_buttons = []
-        for column_name, column_index in columns_to_color.items():
-            column_values = data[:, column_index]
-            node_color = get_node_summary(node_elements, column_values)
-
-            column_color_buttons.append(
-                dict(
-                    args=[{
-                        'marker.color': [None, node_color],
-                        'marker.cmin': [None, np.min(node_color)],
-                        'marker.cmax': [None, np.max(node_color)]
-                    }],
-                    label=column_name,
-                    method='restyle'
-                )
-            )
-        return column_color_buttons
-
-
-def _get_colorscale_buttons(colorscales):
-    colorscale_buttons = []
-    for colorscale in colorscales:
-        colorscale_buttons.append(
+        columns_to_color = range(data.shape[1])
+
+    node_color_map = list(map(lambda x: rgb2hex(get_cmap(colorscale)(x)),
+                              node_colors_color_variable))
+
+    column_color_buttons = [
+        dict(
+            args=[{
+                'marker.color': [None, node_color_map],
+                'marker.cmin': [None, np.min(node_colors_color_variable)],
+                'marker.cmax': [None, np.max(node_colors_color_variable)],
+                'hoverlabel.bgcolor': [None, node_color_map]
+            }],
+            label='color_variable',
+            method='restyle'
+        )
+    ]
+
+    for column in columns_to_color:
+        if is_data_dataframe:
+            column_values = data[column].to_numpy()
+        else:
+            column_values = data[:, column]
+        node_colors = get_node_summary(node_elements, column_values)
+        node_color_map = list(map(lambda x: rgb2hex(get_cmap(colorscale)(x)),
+                                  node_colors))
+
+        column_color_buttons.append(
             dict(
-                args=[{'marker.colorscale': [None, colorscale]}],
-                label=colorscale,
+                args=[{
+                    'marker.color': [None, node_color_map],
+                    'marker.cmin': [None, np.min(node_colors)],
+                    'marker.cmax': [None, np.max(node_colors)],
+                    'hoverlabel.bgcolor': [None, node_color_map]
+                }],
+                label='Column {}'.format(column),
                 method='restyle'
             )
         )
-    return colorscale_buttons
+    return column_color_buttons
+
+
+def _is_array_or_dataframe(color_variable, data):
```

Comment:
It seems to me this function's name (and docstring) does not reflect well its behaviour, as it is also determining whether color_variable is a (fit-)transformer, None, or something else.

Reply:
Good point. I will change it to `_infer_color_variable_kind`